### ISSUE TITLE
Runtime node version for compute app is out of sync with build time node version

### ISSUE THREAD

**Issue by andre-mr** [aws-amplify/amplify-hosting#3671]
### Before opening, please confirm:

- [X] I have checked to see if my question is addressed in the [FAQ](https://github.com/aws-amplify/amplify-hosting/blob/master/FAQ.md).
- [X] I have [searched for duplicate or closed issues](https://github.com/aws-amplify/amplify-hosting/issues?q=is%3Aissue+).
- [X] I have read the guide for [submitting bug reports](https://github.com/aws-amplify/amplify-hosting/blob/master/CONTRIBUTING.md).
- [X] I have done my best to include a minimal, self-contained set of instructions for consistently reproducing the issue.
- [X] I have removed any sensitive information from my code snippets and submission.

### App Id

d3an06sskj66n8

### AWS Region

sa-east-1

### Amplify Hosting feature

Custom build image, Frontend builds, SSR

### Frontend framework

Angular

### Next.js version

13.3.4

### Next.js router

Pages Router

### Describe the bug

Hi.

I customized frontend build to use Node v18, but my Nextjs code is throwing errors, because i'm using node 18+ native functions not supported by v16.

### Expected behavior

Expected to use Node v18.

### Reproduction steps

Hi.

I have customized node version by using "- nvm use 18" preBuild command, according to this answer from @Narrator:
https://github.com/aws-amplify/amplify-hosting/issues/3109#issuecomment-1509264093
This shows in Frontend logs (preBuild phase):
```
[INFO]: # Executing command: nvm use 18
[INFO]: Now using node v18.13.0 (npm v8.19.3)
```

And in another try, according to myself, using custom "node:18.14.2" image:
https://github.com/aws-amplify/amplify-hosting/issues/3109#issuecomment-1454072857
For this i manually checked version:
```
[INFO]: # Executing command: node -v
[INFO]: v18.14.2
[INFO]: # Executing command: npm ci
```

Unfortuntately, none of these is working now.

I added this to one of my pages/api routes:
`console.log(process.version);`

And from Cloudwatch's 'hosting compute logs' i see this:
`v16.20.1`

I wonder if something changed, if that's a bug or I am missing something.
Thank you.

### Build Settings

```yaml
1st approach (-nvm use 18 in default image):
version: 1
frontend:
  phases:
    preBuild:
      commands:
        - nvm use 18
        - npm ci
    build:
      commands:
        - npm run build
  artifacts:
    baseDirectory: .next
    files:
      - '**/*'
  cache:
    paths:
      - node_modules/**/*

2nd approach (custom node:18.14.2 image):
version: 1
frontend:
  phases:
    preBuild:
      commands:
        - node -v
        - npm ci
    build:
      commands:
        - npm run build
  artifacts:
    baseDirectory: .next
    files:
      - '**/*'
  cache:
    paths:
      - node_modules/**/*
```

### Log output

<details>

```
# Put your logs below this line
ReferenceError: structuredClone is not defined

```

</details>

### Additional information
Is Amplify using a different runtime/machine when app is running than when it was being build? all build process is done using Node 18 according to logs, but when we access the app, it is running inside a Node 16 runtime. Is that intended?

_No response_

**Comment by andre-mr** [comment]
wow, this is a serious problem.

installing libraries using node 18 and then running the application using node 16 can cause several incompatibility issues. 

therefore, I strongly suggest users not to use overriding techniques for the runtime, like these:
#3109 _(Fix support for node 18)_ awa
:warning:

and I suggest the team prioritize the upgrade, as node 16 will reach end of life next month, and then we will have applications running on a deprecated runtime. and please make this difference clear in the documentation and in that web console's "build image" screen, since users can make customizations believing that this will be reflected in the runtime.
:astonished:

**Comment by Jay2113** [comment]
Hi @andre-mr 👋 , towards the end of last year (Nov '23) we updated the compute runtime for SSR apps to be in sync with the node version used at build time. Thus, if you are using node 18 at build time via either `nvm use 18` or via live package overrides the compute runtime will similarly use node 18 to serve the SSR requests. This can be verified from the hosting compute logs.

Additionally, we launched a new build image Amazon Linux: 2023 which now supports Node 18 and Node 20 by default: https://github.com/aws-amplify/amplify-hosting/issues/3773#issuecomment-1811288919.
